Tests/Indicators/ImpliedVolatilityTests.cs

```diff
@@ -44,77 +46,138 @@ public void SetUp()
+                var putIndicator = new ImpliedVolatility(put, call, interestRate, dividendYield, model);
+
+                RunTestIndicator(call, put, callIndicator, putIndicator, items, callColumn, putColumn, errorMargin);
+            }
+        }
+
+        // We are unsure about the smoothing function
+        [TestCase("american/third_party_2_greeks.csv")]
+        public void ComparesAgainstExternalDataSingleContractMethod(string subPath, int callColumn = 7, int putColumn = 6, double errorMargin = 10)
+        {
+            var path = Path.Combine("TestData", "greeksindicator", subPath);
+            foreach (var line in File.ReadAllLines(path).Skip(3))
+            {
+                var items = line.Split(',');
+
+                var interestRate = decimal.Parse(items[^2]);
+                var dividendYield = decimal.Parse(items[^1]);
+
+                var model = ParseSymbols(items, path.Contains("american"), out var call, out var put);
+
+                var callIndicator = new ImpliedVolatility(call, interestRate, dividendYield, model);
+                var putIndicator = new ImpliedVolatility(put, interestRate, dividendYield, model);
 
-            var indicator = new ImpliedVolatility(symbol, 0.0530m, 0.0153m);
-            RunTestIndicator(path, indicator, symbol, underlying, errorMargin, column);
+                RunTestIndicator(call, put, callIndicator, putIndicator, items, callColumn, putColumn, errorMargin);
+            }
         }
 
-        // For comparing IB's value
-        [TestCase("SPX230811C04300000", 0.2)]
-        [TestCase("SPX230811C04500000")]
-        [TestCase("SPX230811C04700000")]
-        [TestCase("SPX230811P04300000", 0.015)]
-        [TestCase("SPX230811P04500000")]
-        [TestCase("SPX230811P04700000", 0.075)]
-        [TestCase("SPX230901C04300000")]
-        [TestCase("SPX230901C04500000")]
-        [TestCase("SPX230901C04700000")]
-        [TestCase("SPX230901P04300000")]
-        [TestCase("SPX230901P04500000")]
-        [TestCase("SPX230901P04700000")]
-        [TestCase("SPY230811C00430000", 0.055)]
-        [TestCase("SPY230811C00450000", 0.015)]
-        [TestCase("SPY230811C00470000")]
-        [TestCase("SPY230811P00430000", 0.015)]
-        [TestCase("SPY230811P00450000")]
-        [TestCase("SPY230811P00470000", 0.075)]
-        [TestCase("SPY230901C00430000", 0.025)]
-        [TestCase("SPY230901C00450000")]
-        [TestCase("SPY230901C00470000")]
-        [TestCase("SPY230901P00430000")]
-        [TestCase("SPY230901P00450000")]
-        [TestCase("SPY230901P00470000", 0.035)]
-        public void ComparesAgainstExternalDataAfterReset(string fileName, double errorMargin = 0.01, int column = 2)
+        [TestCase("american/third_party_1_greeks.csv")]
+        public void ComparesAgainstExternalDataAfterReset(string subPath, int callColumn = 7, int putColumn = 6, double errorMargin = 0.08)
```

Comment:
- This test is very similar to the one above, reset could be a test case argument, true/false so it's a single unit tests
- Should replace `decimal.Parse` by the `Parse.Decimal` extensions we have to avoid culture specific